idunn/datasources/pages_jaunes.py

```diff
@@ -57,6 +60,33 @@ def format_where(bbox):
         left, bot, right, top = bbox
         return f"gZ{left:.6f},{bot:.6f},{right:.6f},{top:.6f}"
 
+    @classmethod
+    def fetch_search(cls, query: SearchQueryParams, intentions=None, is_france_query=False):
+        if len(intentions) > 0:
+            return asyncio.create_task(
+                run_in_threadpool(
+                    pj_source.search_places,
+                    query,
+                    intentions[0].description._place_in_query,
+                ),
+                name="ia_fetch_pj",
+            )
+        return asyncio.create_task(
```

Comment:
I'd argue that the caller should be responsible of calling `asyncio.create_task` which implies non-trivial side effects. Then we could also make this method async to make it more obvious that it returns an awaitable.

This remark also applies to other `fetch_search` methods.

Reply:
yes indeed, i need to change that part